internal/reminder/reminder.go

```diff
@@ -132,28 +133,46 @@ func (r *reminder) Stop() {
 	})
 }
 
-func (r *reminder) sendReminders(ctx context.Context) []error {
+func (r *reminder) sendReminders(ctx context.Context) error {
 	logger := zerolog.Ctx(ctx)
 
 	// Fetch a batch of repositories
 	repos, err := r.getRepositoryBatch(ctx)
 	if err != nil {
-		logger.Error().Err(err).Msg("unable to fetch repositories")
-		return []error{err}
+		return fmt.Errorf("error fetching repository batch: %w", err)
+	}
+
+	if len(repos) == 0 {
+		logger.Debug().Msg("no repositories to send reminders for")
+		return nil
 	}
 
 	logger.Info().Msgf("created repository batch of size: %d", len(repos))
 
-	// Update the reminder_last_sent for each repository to export as metrics
+	messages, err := createReminderMessages(ctx, repos)
+	if err != nil {
+		return fmt.Errorf("error creating reminder messages: %w", err)
+	}
+
+	err = r.eventPublisher.Publish(events.TopicQueueRepoReminder, messages...)
+	if err != nil {
+		return fmt.Errorf("error publishing messages: %w", err)
+	}
+
+	repoIds := make([]uuid.UUID, len(repos))
 	for _, repo := range repos {
-		logger.Debug().Str("repo", repo.ID.String()).
-			Time("previously", repo.ReminderLastSent.Time).Msg("updating reminder_last_sent")
-		err := r.store.UpdateReminderLastSentById(ctx, repo.ID)
-		if err != nil {
-			logger.Error().Err(err).Str("repo", repo.ID.String()).Msg("unable to update reminder_last_sent")
-			return []error{err}
-		}
-		// TODO: Send the actual reminders
+		repoIds = append(repoIds, repo.ID)
+	}
+
+	// TODO: Collect Metrics
+	// Potential metrics:
+	// - Gauge: Number of reminders in the current batch
+	// - UpDownCounter: Average reminders sent per batch
+	// - Histogram: reminder_last_sent time distribution
+
+	err = r.store.UpdateReminderLastSentForRepositories(ctx, repoIds)
+	if err != nil {
+		return fmt.Errorf("reminders published but error updating last sent time: %w", err)
```

Comment:
Bulk updates for all repos is fine, IMO. We don't want to have a `ms` difference. The only value of this metric is in minutes/hours to make sure reconciliation is happening at a regular interval and we can reconcile every repo in like 24 hours or so.